mint-client/src/mint/mod.rs

```diff
@@ -94,22 +94,6 @@ impl<'c> MintClient<'c> {
         Ok(coins)
     }
 
-    // TODO: implement input generation with change to avoid error on missing coin denominations
-    /// Select coins to fund a transaction with.
-    ///
-    /// **ATTENTION**: calling this function multiple times without committing the batch to the
-    /// database is not supported and will result in an accidental double spend.
-    pub fn create_coin_input(
-        &self,
-        mut batch: BatchTx,
-        amount: Amount,
-    ) -> Result<(Vec<KeyPair>, Coins<Coin>)> {
-        let coins = self.select_and_spend_coins(batch.subtransaction(), amount)?;
-        let (spend_keys, coins) = self.create_coin_input_from_coins(coins)?;
-        batch.commit();
-        Ok((spend_keys, coins))
-    }
```

Comment:
I ended up removing this from the mint client because it's not really usable given that it only works with exact change.